src/Microsoft.AspNetCore.Blazor.Build/targets/Blazor.MonoRuntime.targets

```diff
@@ -351,7 +351,10 @@
   <Target
       Name="_LinkBlazorApplication"
       Condition="$(_BlazorShouldLinkApplicationAssemblies) != ''"
-      Inputs="$(BlazorBuildLinkerInputsCache)"
+      Inputs="$(BlazorBuildLinkerInputsCache);
+              @(IntermediateAssembly);
+              @(ReferenceCopyLocalPaths->WithMetadataValue('Extension','.dll'));
```

Comment:
The extra metadata check here probably isn't needed. Is there a particular problem you're trying to avoid?

Reply:
I don't care about the PDBs

Reply:
Ah interesting. Hadn't thought about that 👍 

The case where Razor uses assemblies we're OK with ref assemblies so we don't have to deal with this.